### PR TITLE
chore(api, worker, dal): remove `SubscriberPreferenceRepository`

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* remove `SubscriberPreferenceRepository` and all references
  * all subscriber preferences are now stored in the `Preferences` collection

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/migrations/preference-centralization/preference-centralization-migration.spec.ts [hunk 1]
```diff
@@ -1,201 +1,7 @@
...
-    );
-
-    expect(workflowPreference?.preferences).to.deep.equal(DEFAULT_WORKFLOW_PREFERENCES);
-    expect(userPreference?.preferences).to.deep.equal(DEFAULT_WORKFLOW_PREFERENCES);
-  });
+  /**
+   * IMPORTANT: This migration depends on SubscriberPreferencesRepository which is now removed.
+   * Please checkout the `v2.1.0` tag and run the migration from there.
+   * @see https://github.com/novuhq/novu/releases/tag/v2.1.0
+   */
```
Comment (rifont) [inline apps/api/migrations/preference-centralization/preference-centralization-migration.spec.ts:6]: This migration doesn't work without the `SubscriberPreferencesRepository`, so we leave a note to checkout the relevant tag and run the migration from there before proceeding.

File: apps/api/migrations/preference-centralization/preference-centralization-migration.ts
Comment (rifont) [inline apps/api/migrations/preference-centralization/preference-centralization-migration.ts:1]: Ditto, leaving a note to checkout the relevant commit.

File: apps/api/migrations/subscriber-preferences-level/subscriber-preferences-level.migration.ts
Comment (rifont) [inline apps/api/migrations/subscriber-preferences-level/subscriber-preferences-level.migration.ts:1]: This was a very old migration to add the `level` flag, and it's not referenced in the [migration docs](https://docs.novu.co/additional-resources/data-migrations) so I've just deleted it.

Comment (rifont) [inline apps/api/migrations/subscriber-preferences-level/subscriber-preferences-level.migration.ts:1]: This was a very old migration to add the `level` flag the to subscriber preference collection records. It's not referenced in the [migration docs](https://docs.novu.co/additional-resources/data-migrations) so I've just deleted it.

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**